Do not create blob temp folder for ephemeral sessions

youennf opened the pull request:
#### e3e5b70c243a9f274e584c14f45f3704327f2b9d
<pre>
Do not create blob temp folder for ephemeral sessions
<a href="https://bugs.webkit.org/show_bug.cgi?id=257950">https://bugs.webkit.org/show_bug.cgi?id=257950</a>
rdar://110545873

Reviewed by Sihui Liu.

We are creating a temp folder for blobs to use mapped memory to store blobs, as a way to limit networking process jetsams.
This creation was not dependent on session being ephemeral or not.
We are now skipping the creation of this temp folder for ephemeral sessions.
This will trigger use of regular memory for storing blobs, which may increase networking process jetsams.

* Source/WebKit/NetworkProcess/cocoa/NetworkSessionCocoa.mm:
(WebKit::NetworkSessionCocoa::NetworkSessionCocoa):

Canonical link: <a href="https://commits.webkit.org/265127@main">https://commits.webkit.org/265127@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2b2d4e80a4284be0411b26a49933cc6a0cebdde3

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/9681 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/9932 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/10177 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/11344 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/9458 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/11918 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/9891 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/12380 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/9835 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/10679 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/8241 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/11503 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/7983 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/16206 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/9081 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/8952 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/12284 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/9442 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/7700 "6 failures") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/8625 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2379 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/12850 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/9212 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

szewai reviewed (commented):
Is this code run on both macOS and iOS? It seems network process currently only has access to temp folder on iOS (see `NetworkProcess::addWebsiteDataStore`), so only creating temp folder in macOS will fail (even for persistent session).

szewai reviewed (approved):
r=me

youennf committed:
Do not create blob temp folder for ephemeral sessions
https://bugs.webkit.org/show_bug.cgi?id=257950
rdar://110545873

Reviewed by Sihui Liu.

We are creating a temp folder for blobs to use mapped memory to store blobs, as a way to limit networking process jetsams.
This creation was not dependent on session being ephemeral or not.
We are now skipping the creation of this temp folder for ephemeral sessions.
This will trigger use of regular memory for storing blobs, which may increase networking process jetsams.

* Source/WebKit/NetworkProcess/cocoa/NetworkSessionCocoa.mm:
(WebKit::NetworkSessionCocoa::NetworkSessionCocoa):

Canonical link: https://commits.webkit.org/265127@main
